### PR TITLE
Use alpine for fluentd

### PR DESCRIPTION
**Pull request by nickgerace**
Relevant issue: https://github.com/rancher/rancher/issues/33390

## Image Size Differences

```
(new)        rancher/fluentd    dev         96MB
(current)    rancher/fluentd    v0.1.24    166MB
```

## Why Alpine?

vs. | reasoning
--- | ---
SLE BCI | manual ruby installation requires version management (Ruby has both compile time and runtime dependencies)
Debian | same as SLE BCI (manual is required since it has old Ruby packages)
Ubuntu | same as SLE BCI (manual is required since it has old Ruby packages and PPAs are a third party dependency)
Ruby | meant for builds, has vulnerabilities
Alpine | up to date Ruby installation with ENVs, libraries, PATH, etc. configured properly along with a smaller image size

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -19,34 +19,20 @@
...
-# 3. Remove build dependencies.
-# 4. Cleanup leftover caches & files.
-# RUN BUILD_DEPS="make gcc g++ libc6-dev ruby-dev libffi-dev" \
-RUN BUILD_DEPS="make ruby-dev build-essential autoconf automake libtool libsnappy-dev" \
-    && clean-install $BUILD_DEPS \
-                     ca-certificates \
-                     libjemalloc1 \
-                     ruby \
+RUN apk update \
+    && apk add --no-cache ca-certificates ruby ruby-dev ruby-irb ruby-etc ruby-webrick build-base \
```
Comment (paynejacob) [inline package/Dockerfile:28]: are all of these required at runtime? could we put some of them in a virtual package and remove them at the end?

Comment (nickgerace) [reply package/Dockerfile:28]: Yes. Ruby has both compile time and runtime dependencies.

**Review by paynejacob — APPROVED** [review]

File: package/Dockerfile [hunk 2]
```diff
@@ -19,34 +19,22 @@
...
     && echo 'gem: --no-document' >> /etc/gemrc \
-    && gem install bundle -v 0.0.1 \
+    && gem install bundler \
     && bundle install \
```
Comment (rmweir) [inline package/Dockerfile:31]: Would using strict version here prevent random issues in the future?

Comment (nickgerace) [reply package/Dockerfile:31]: We could do this, but bundler is the bootstrapper for the gems themselves, so leaving it unversioned should be fine. I'd be happy to change it though if it's a non-starter.

**Review by paynejacob — CHANGES_REQUESTED** [review]
dockerfile changes looks good but the build is failing

**Comment by nickgerace** [comment]
Just so reviewers are aware: https://github.com/rancher/rancher/issues/33486

**Comment by nickgerace** [comment]
This has been unblocked by @rosskirkpat's work on https://github.com/rancher/rancher/issues/33486.

**Review by paynejacob — APPROVED** [review]

**Pull request merged**